otx/algorithms/action/adapters/mmaction/data/cls_dataset.py

```diff
@@ -0,0 +1,144 @@
+            During iterating DatasetsetEntitiy, this fucnction generates video_info(dictionary)
+            video_info records metadata for each video, and it contains
+                - total_frame: Total frame number of the video, this value will be used to sample frames for training
+                - start_index: Offset for the video, this value will be added to sampled frame indices for the video
+                - label: Action category of the video
+                - modality = Modality of data, 'RGB' or 'Flow(Optical Flow)'
+            """
+            video_info = {}
+            start_index = 0
+            for idx, item in enumerate(self.otx_dataset):
+                metadata = item.get_metadata()[0].data
+                if metadata.video_id in video_info:
+                    video_info[metadata.video_id]["total_frames"] += 1
+                    video_info[metadata.video_id]["start_index"] = start_index
+                else:
+                    if len(item.get_annotations()) > 0:
+                        label = int(item.get_roi_labels(self.labels)[0].id)
+                    else:
+                        label = None
+                    ignored_labels = np.array([self.label_idx[lbs.id] for lbs in item.ignored_labels])
+                    video_info[metadata.video_id] = {
+                        "total_frames": 1,
+                        "start_index": idx,
+                        "label": label,
+                        "ignored_labels": ignored_labels,
+                        "modality": self.modality,
+                    }
+                    start_index = idx
+
+            self.video_info.update(video_info)
+
+        def __getitem__(self, index: int):
+            """Prepare training data item.
+
+            Action classification needs video for training, therefore this function generate item from video_info
+            """
+
+            item = self.video_info[list(self.video_info.keys())[index]]
+            return item
+
+    @check_input_parameters_type({"otx_dataset": DatasetParamTypeCheck})
+    # pylint: disable=too-many-arguments, invalid-name, super-init-not-called
+    # TODO Check need for additional params such as multi_class, with_offset
+    def __init__(
+        self,
+        otx_dataset: DatasetEntity,
+        labels: List[LabelEntity],
+        pipeline: Sequence[dict],
+        test_mode: bool = False,
+        modality: str = "RGB",  # [RGB, FLOW(Optical flow)]
+    ):
+        self.otx_dataset = otx_dataset
+        self.labels = labels
+        self.test_mode = test_mode
+        self.modality = modality
+
+        self.video_infos = OTXRawframeDataset._DataInfoProxy(otx_dataset, labels, modality)
+
+        self.pipeline = Compose(pipeline)
+        for pip in self.pipeline.transforms:
+            if isinstance(pip, RawFrameDecode):
+                pip.otx_dataset = self.otx_dataset
```

Comment:
Maybe `transform` could be used instead of `pip` since we are going over `.transforms` in the loop?
```suggestion
        for transform in self.pipeline.transforms:
            if isinstance(transform, RawFrameDecode):
                transform.otx_dataset = self.otx_dataset
```

Reply:
or `pipeline` maybe? whichever sounds logically correct.

Reply:
Convert to transform Thanks